### PR TITLE
Generalize the 2020 regression report to show local unexpected failures

### PR DESCRIPTION
**Pull request by SimonSapin**
Example usage:

```
./mach test-wpt --release --layout-2020 --log-raw /tmp/servo.log
./tests/wpt/reftests-report/gen.py /tmp/servo.log
firefox ./tests/wpt/reftests-report/report.html
```

Produces a report similar https://community-tc.services.mozilla.com/api/index/v1/task/project.servo.layout-2020-regressions-report/artifacts/public/regressions.html, but showing unexpected reftest failures. The CI-generated one shows Layout 2020 failures (expected or not) that succeed in Layout 2013.

### CODE REVIEW

**Comment by jdm** [comment]
```
# Push to master:
Found task <from index> indexed at project.servo.<MagicMock name='mock.check_output().decode().strip().__radd__()' id='140578698698592'>
Found task <from index> indexed at project.servo.docker-image.263b6783749c63391a9ca609cff096214027bfb21b16c61e851197f068cd8def
Scheduled b'<new id>': Linux x64: Upload docs to GitHub Pages
Traceback (most recent call last):
  File "./etc/taskcluster/mock.py", line 61, in <module>
    decision_task.main("github-push")
  File "/repo/etc/taskcluster/decision_task.py", line 15, in main
    tasks(task_for)
  File "/repo/etc/taskcluster/decision_task.py", line 71, in tasks
    function()
  File "/repo/etc/taskcluster/decision_task.py", line 268, in layout_2020_regressions_report
    .with_index_at("layout-2020-regressions-report")
AttributeError: 'NoneType' object has no attribute 'create'
Name                               Stmts   Miss  Cover   Missing
----------------------------------------------------------------
etc/taskcluster/decision_task.py     184     36    80%   73-98, 149, 287-293, 347, 394, 398, 414, 443, 462-463, 690-697, 801
etc/taskcluster/decisionlib.py       318      6    98%   77-78, 471-472, 485-486
etc/taskcluster/mock.py               38      7    82%   63-72
----------------------------------------------------------------
TOTAL                                540     49    91%
[taskcluster 2020-05-06 00:15:14.434Z] === Task Finished ===
```

File: etc/taskcluster/decisionlib.py [hunk 1]
```diff
@@ -157,6 +157,9 @@ def __init__(self, name):
 
     with_extra = chaining(update_attr, "extra")
 
+    def with_index_at(self, index_path):
+        self.routes.append("index.%s.%s" % (CONFIG.index_prefix, index_path))
```
Comment (jdm) [inline etc/taskcluster/decisionlib.py:161]: This needs to return self.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by SimonSapin** [comment]
Indeed, that fixes running `mock.py`. Thanks!

@bors-servo r=jdm

**Pull request merged**